docs/topics/upgrade.md

```diff
@@ -0,0 +1,77 @@
+
+To get the list of all available Kubernetes versions and upgrades, run the `orchestrators` command and specify Kubernetes orchestrator type:
+
+```bash
+./bin/aks-engine orchestrators --orchestrator Kubernetes
+```
+
+To get the information specific to the cluster, provide its current Kubernetes version in the `version` arg:
+```bash
+./bin/aks-engine orchestrators --orchestrator Kubernetes --version 1.11.5
+```
+
+4) If using `aks-engine upgrade` in production, it is recommended to run a dry run test on an identical cluster before performing the upgrade, especially if the cluster configuration is "unusual", or in other words differs significantly from defaults. The reason for this is that `aks-engine` supports many different cluster configurations and the extent of E2E testing that the AKS-Engine team runs cannot possibly cover every single configuration out there. Therefore, it is recommended that you make sure that your specific cluster configuration works with the existing upgrade implementation before starting this long-running operation.
+
+5) `aks-engine upgrade` is backwards compatible. If you deployed with `aks-engine` version `0.27.x`, you can run upgrade with version `0.29.y`. In fact, it is recommended that you use the latest available `aks-engine` version when running an upgrade operation. This will ensure that you get the latest available software and bug fixes in your upgraded cluster.
+
+### Under the hood
+
+During the upgrade, *aks-engine* successively visits virtual machines that constitute the cluster (first the master nodes, then the agent nodes) and performs the following operations:
+ - cordon the node and drain existing workload
+ - delete the VM
+ - create new VM and install desired orchestrator version
+ - add the new VM to the cluster
+
+### Simple steps to run upgrade
+
+Once you have read all the [prerequirements](#pre-requirements), run `aks-engine upgrade` with the appropriate arguments:
+
+```bash
+./bin/aks-engine upgrade \
+  --subscription-id <subscription id> \
+  --deployment-dir <aks-engine output directory > \
+  --location <resource group location> \
+  --resource-group <resource group name> \
+  --upgrade-version <desired Kubernetes version> \
+  --auth-method client_secret \
+  --client-id <service principal id> \
+  --client-secret <service principal secret>
+```
+For example,
+```bash
+./bin/aks-engine upgrade \
+  --subscription-id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx \
+  --deployment-dir ./_output/test \
+  --location westus \
+  --resource-group test-upgrade \
+  --upgrade-version 1.8.7 \
+  --client-id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx \
+  --client-secret xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+```
+
+### What can go wrong
+
+By its nature, the upgrade operation is long running and potentially could fail for various reasons, such as temporary lack of resources, etc. In this case, rerun the command. The `upgrade` command is idempotent, and will pick up execution from the point it failed on.
```

Comment:
Would this be something handy to put in the [troubleshooting docs](https://github.com/Azure/aks-engine/blob/master/docs/howto/troubleshooting.md) as well?

Reply:
Possibly. Maybe I should just add a "failed upgrade" in the troubleshooting doc that redirects to this page instead of repeating this section.